tests/subsys/logging/log_stack/src/test.inc

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <logging/log.h>
+
+#define MODULE_NAME test
+#define CXX_MODULE_NAME test_cxx
+
+/* The idea of this test is to estimate stack usage of logging messages. Each
+ * test checks stack usage before and after logging a message. Increase of stack
+ * usage is interpreted as stack used by the logging. Test has been used to
+ * characterize some common architectures and it will fail if more stack is
+ * used.
+ *
+ * This will help to detect when changes in logging, toolchain or modules used
+ * by logging increase stack usage. Failure of the test should lead to investigation
+ * to determine if stack usage change is justified. If it is justified, thresholds
+ * may be updated.
+ */
+
+#if __cplusplus
+LOG_MODULE_REGISTER(CXX_MODULE_NAME);
+#else
+LOG_MODULE_REGISTER(MODULE_NAME);
+#endif
+
+#if __cplusplus
+#define TEST_SUFFIX cxx
+#else
+#define TEST_SUFFIX cc
+#endif
+
+#define SUFFIXED_NAME(_name) UTIL_CAT(_name, UTIL_CAT(_, TEST_SUFFIX))
+
+#define WRAP_TEST(suite_name, test_name)                   \
+	ZTEST(SUFFIXED_NAME(suite_name), SUFFIXED_NAME(test_name)) \
+	{                                                  \
+		test_name();                               \
+	}
```

Comment:
If we didn't have main.c and main.cxx in the same executable we wouldn't have to generate the symbol names like this would we?

I propose we use testcase.yaml to determine whether either main.c or maincxx.cxx is used so we don't have to deal with this problem.

Reply:
Done